src/bindgen/ir/constant.rs

```diff
@@ -152,20 +155,44 @@ impl Constant {
             return Err("Unhanded const definition".to_owned());
         }
 
-        Ok(Constant {
-            name: name,
+        Ok(Constant::new(
+            path,
+            ty,
+            LiteralExpr::load(&item.expr)?,
+            Cfg::append(mod_cfg, Cfg::load(&item.attrs)),
+            AnnotationSet::load(&item.attrs)?,
+            Documentation::load(&item.attrs),
+        ))
+    }
+
+    pub fn new(
+        path: Path,
+        ty: Type,
+        value: LiteralExpr,
+        cfg: Option<Cfg>,
+        annotations: AnnotationSet,
+        documentation: Documentation,
+    ) -> Self {
+        let export_name = path.name().to_owned();
+        Self {
+            path: path,
```

Comment:
I think it might be more idiomatic rust to change `path: path` to `path` in this constructor as they all have the same name. Ditto for all the other field initializers.